examples/module/src/gizmo/my_gizmo.py

```diff
@@ -28,6 +28,18 @@ def new(cls, config: ComponentConfig, dependencies: Mapping[ResourceName, Resour
         gizmo.my_arg = config.attributes.fields["arg1"].string_value
         return gizmo
 
+    @classmethod
+    def validate_config(cls, config: ComponentConfig) -> List[str]:
+        # Custom validation can be done by specifiying a validate function like this one. Validate functions
+        # can throw `throw` errors that will be returned to the parent through gRPC. Validate functions can
+        # also return a vector of strings representing the implicit dependencies of the resource.
+        if "invalid" in config.attributes.fields:
+            raise Exception(f"'invalid' attribute not allowed for model {cls.SUBTYPE}:{cls.MODEL}")
+        motor = [config.attributes.fields["motor"].string_value]
+        if motor == []:
+            raise Exception("A motor is required for Gizmo module.")
+        return motor
```

Comment:
Can we also add a validation that `arg1` exists in the config? Since that is also required?